### PR TITLE
✨Adding argument commands to Symply

### PR DESCRIPTION
**Pull request by AzaleeX**
**Add all the arguments:**
- `TargetArgument`
- `StringEnumArgument`
- `RawTextArgument`
- `RawStringArgument`
- `PositionArgument`
- `IntegerArgument`
- `FloatArgument`
- `BooleanArgument`
- ` BlockArgument`

**Fixing files with PHP Cs Fixer**

Ref: #10

### CODE REVIEW

**Review by Virvolta — APPROVED** [review]
:+1:

File: symply/command/arguments/CommandArgument.php [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+abstract class CommandArgument
+{
+
+	protected CommandParameter $parameter;
+
+	/** @param string $name */
+	private string $name;
+
+	/** @param bool $optional */
+	private string $optional;
```
Comment (SunsetTadpole78) [inline symply/command/arguments/CommandArgument.php]: useless variable documentations

File: symply/command/arguments/CommandArgument.php [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+		$this->name = $name;
+		$this->optional = $optional;
+
+		$this->parameter = new CommandParameter();
+		$this->parameter->paramName = $name;
+		$this->parameter->paramType = AvailableCommandsPacket::ARG_FLAG_VALID; $this->parameter->paramType |= $this->getNetworkType();
+		$this->parameter->isOptional = $this->isOptional();
+	}
+
+	abstract public function canExecute(string $testString, CommandSender $sender) : bool;
```
Comment (SunsetTadpole78) [inline symply/command/arguments/CommandArgument.php]: rename to `isValid()` ?

File: symply/command/arguments/list/BlockCommandArgument.php [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+	}
+
+	public function execute(string $argument, CommandSender $sender) : Item|bool|Enchantment|Effect{
+		$argument = str_starts_with($argument, "minecraft:") ? $argument : (str_contains($argument, ":") ? $argument : "minecraft:" . $argument);
+		$blockIdentifier = StringToItemParser::getInstance()->parse($argument);
+
+		if ($blockIdentifier != null) {
+			return $blockIdentifier;
+		}
+		return false;
```
Comment (SunsetTadpole78) [inline symply/command/arguments/list/BlockCommandArgument.php]:
```suggestion

		return StringToItemParser::getInstance()->parse($argument) ?? false;
```

File: symply/command/arguments/list/BooleanCommandArgument.php [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+namespace symply\command\arguments\list;
+
+use pocketmine\command\CommandSender;
+
+class BooleanCommandArgument extends StringEnumCommandArgument
+{
+	/**
+	 * @var array
+	 */
```
Comment (SunsetTadpole78) [inline symply/command/arguments/list/BooleanCommandArgument.php]: specify array content

File: symply/command/arguments/list/BooleanCommandArgument.php [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+	const VALUES = [
+		"true" && "t" => true,
+		"false" && "f" => false,
```
Comment (SunsetTadpole78) [inline symply/command/arguments/list/BooleanCommandArgument.php]:
```suggestion
		"false" && "f" => false
```

File: symply/command/arguments/list/BooleanCommandArgument.php [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+	];
+
+	public function getTypeName() : string{
+		return "bool";
+	}
+
+	public function execute(string $argument, CommandSender $sender) : mixed{
```
Comment (SunsetTadpole78) [inline symply/command/arguments/list/BooleanCommandArgument.php]: specify return value

File: symply/command/arguments/list/PositionCommandArgument.php [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+
+	public function getTypeName() : string{
+		return "x y z";
+	}
+
+	public function canExecute(string $testString, CommandSender $sender) : bool{
+		$position = explode(" ", $testString);
+		if (count($position) === 3) {
+			foreach($position as $vector) {
+				if ($this->isValid($vector, $sender instanceof Vector3)) {
```
Comment (SunsetTadpole78) [inline symply/command/arguments/list/PositionCommandArgument.php]: `CommandSender` will never be a  `Vector3`

File: symply/command/arguments/list/PositionCommandArgument.php [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+					return false;
```
Comment (SunsetTadpole78) [inline symply/command/arguments/list/PositionCommandArgument.php:55]: return false if it is valid?

File: symply/command/arguments/list/StringEnumCommandArgument.php [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+		$this->parameter->enum = new CommandEnum("", $this->getEnumValues());
+	}
+
+	public function getNetworkType() : int{
+		return -1;
+	}
+
+	/**
+	 * @return mixed
+	 */
```
Comment (SunsetTadpole78) [inline symply/command/arguments/list/StringEnumCommandArgument.php]: invalid & useless comment

File: symply/command/arguments/list/StringEnumCommandArgument.php [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+	public function canExecute(string $testString, CommandSender $sender) : bool{
+		return (bool) preg_match(
+			"/^(" . implode("|", array_map("\\strtolower", $this->getEnumValues())) . ")$/iu",
+			$testString
+		);
+	}
+
+	/**
+	 * @return mixed
+	 */
```
Comment (SunsetTadpole78) [inline symply/command/arguments/list/StringEnumCommandArgument.php]: useless comment

File: symply/command/arguments/list/StringEnumCommandArgument.php [hunk 11]
```diff
@@ -0,0 +1,74 @@
...
+	public function getValue(string $string){
```
Comment (SunsetTadpole78) [inline symply/command/arguments/list/StringEnumCommandArgument.php]: add return value type

File: symply/command/arguments/list/TargetCommandArgument.php [hunk 12]
```diff
@@ -0,0 +1,53 @@
...
+	public function getTypeName() : string {
+		return "target";
+	}
+
+	public function canExecute(string $testString, CommandSender $sender) : bool {
+		return (bool) preg_match("/^(?!rcon|console)[a-zA-Z0-9_ ]{1,16}$/i", $testString);
+	}
+
+	public function execute(string $argument, CommandSender $sender) : string {
+		return strtolower($argument);
```
Comment (SunsetTadpole78) [inline symply/command/arguments/list/TargetCommandArgument.php]: return the player if he is online ?

File: symply/events/session/SessionErrorEvent.php
Comment (SunsetTadpole78) [inline symply/events/session/SessionErrorEvent.php:1]: change not related to pull request

File: symply/events/session/SessionPingUpdateEvent.php
Comment (SunsetTadpole78) [inline symply/events/session/SessionPingUpdateEvent.php:1]: change not related to pull request

File: symply/plugin/listener/PacketListener.php
Comment (SunsetTadpole78) [inline symply/plugin/listener/PacketListener.php:1]: change not related to pull request

File: symply/utils/PacketUtils.php
Comment (SunsetTadpole78) [inline symply/utils/PacketUtils.php:1]: change not related to pull request

**Review by SunsetTadpole78 — CHANGES_REQUESTED** [review]
Some points to change and test.

File: symply/command/arguments/list/TargetCommandArgument.php (hunk 12 above)
Comment (SenseiTarzan) [reply symply/command/arguments/list/TargetCommandArgument.php]: Send player class if he is online or string if he is offline

**Comment by AzaleeX** [comment]
Hey there @SunsetTadpole78, I was finally able to do everything you want me to modify in this pull request if you'd like to take a look.

File: symply/command/arguments/CommandArgument.php [hunk 13]
```diff
@@ -0,0 +1,70 @@
...
+use pocketmine\command\CommandSender;
+use pocketmine\network\mcpe\protocol\AvailableCommandsPacket;
+use pocketmine\network\mcpe\protocol\types\command\CommandParameter;
+
+abstract class CommandArgument
+{
+
+	protected CommandParameter $parameter;
+	private string $name;
+	private string $optional;
```
Comment (SunsetTadpole78) [inline symply/command/arguments/CommandArgument.php:38]: wrong variable type

File: symply/command/arguments/list/TargetCommandArgument.php [hunk 14]
```diff
@@ -0,0 +1,62 @@
...
+
+	public function execute(string $argument, CommandSender $sender) : string|Player {
+		$target = strtolower($argument);
+
+		$player = Server::getInstance()->getPlayerExact($target);
+		if ($player instanceof Player) {
+			return $player;
+		} else {
+			return $target;
+		}
```
Comment (SenseiTarzan) [inline symply/command/arguments/list/TargetCommandArgument.php]: change this for 
```php
return Server::getInstance()->getPlayerExact($target) ?? $target;
```

**Review by SenseiTarzan — APPROVED** [review]

**Pull request closed without merging**